### PR TITLE
chore: fix conda badge and update dependabot

### PR DESCRIPTION
**Pull request by Saransh-cpp**
- Currently, the conda badge points towards decay language
- Update `dependabot.yml` to catch next major versions
- Update `setup-python` with python version for `v4`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
```
Comment (Saransh-cpp) [inline .github/workflows/ci.yml:25]: @henryiii should we specify the python version, or should we lock this to `v3.0.0` and wait for the maintainers to make it optional again?

Comment (henryiii) [reply .github/workflows/ci.yml:25]: I think we should start specifying `python-version: 3.x` and avoid getting locked to `setup-python/v3`. I'm hoping that https://github.com/actions/setup-python/issues/421 would get resolved by restoring the default behavior, but nothing has happened there yet.

Comment (henryiii) [reply .github/workflows/ci.yml:25]:
```suggestion
      with:
        python-version: 3.x
```

File: README.md [hunk 2]
```diff
@@ -623,8 +623,8 @@ Support for this work was provided by the National Science Foundation cooperativ
 [black-link]:               https://github.com/psf/black
 [codecov-badge]:            https://codecov.io/gh/scikit-hep/vector/branch/main/graph/badge.svg?token=YBv60ueORQ
 [codecov-link]:             https://codecov.io/gh/scikit-hep/vector
-[conda-version]:            https://img.shields.io/conda/vn/conda-forge/decaylanguage.svg
-[conda-link]:               https://github.com/conda-forge/decaylanguage-feedstock
+[conda-version]:            https://img.shields.io/conda/vn/conda-forge/vector.svg
+[conda-link]:               https://github.com/conda-forge/vector-feedstock
```
Comment (henryiii) [inline README.md:627]: 😳

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Saransh-cpp) [reply .github/workflows/ci.yml]: Thanks!

File: README.md (hunk 2 above)
Comment (Saransh-cpp) [reply README.md:627]: All the discussions about badges paid off 🤣

**Pull request merged**